var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -169,6 +169,10 @@ class Hpx(CMakePackage, CudaPackage, ROCmPackage):
 
     # Patches and one-off conflicts
 
+    # Asio 1.34.0 removed io_context::work, used by HPX:
+    # https://github.com/chriskohlhoff/asio/commit/a70f2df321ff40c1809773c2c09986745abf8d20.
+    conflicts("^asio@1.34:")
```

Comment:
@hkaiser FYI: asio 1.34.0 breaks HPX because it removes the deprecated `io_context::work` (https://github.com/chriskohlhoff/asio/commit/a70f2df321ff40c1809773c2c09986745abf8d20). I haven't checked if there are more things that HPX depends on that were removed. Build logs here: https://gitlab.spack.io/spack/spack/-/pipelines/1017781.

I've now added a conflict for any version of HPX, but if this is updated for HPX 1.11.0 this could be changed to:
```suggestion
    # Asio 1.34.0 removed io_context::work, used by HPX:
    # https://github.com/chriskohlhoff/asio/commit/a70f2df321ff40c1809773c2c09986745abf8d20.
    conflicts("^asio@1.34:", when="@:1.10")
```

Reply:
Thanks for the heads-up. I'll investigate.

Reply:
See also https://github.com/STEllAR-GROUP/hpx/issues/6648

Reply:
Given https://github.com/STEllAR-GROUP/hpx/pull/6649 will most likely be in HPX 1.11, I added the upper bound on the conflict here. Thanks @hkaiser for the quick fix.

Reply:
Thanks  @msimberg !